### PR TITLE
ChainEvents Polling and HTTP providers

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #3028 

## Description of Changes
- Refactors EVM chain events listeners to poll as block time intervals with address/event filters
   - Main changes introduced int CE `subscriber.ts` files to fetch filtered block logs then use traditional enrich/process flow
- Creates `EVMSourceMap` type in CE which allows for mapping of contract -> relevant events to listen for 
- Updates unit/integration tests to work with and test polling scenarios 
- Updates all `balanceType = 'ethereum` ChainNodes to use https rpc endpoints for event polling and cross-app
- Updates usage of Web3 providers to prefer https but fallback to wss if needed 
- CW app updated everywhere WSSProviders used  

## Test Plan
- Run CE Unit + Integration tests
- Run CE on staging 

## Deployment Plan

1. Includes a migration for white glove style updates to existing evm endpoints. Migration should run first(CW) before code changes are deployed
2. Follow typical CE deployment and monitoring plans 

## Other Considerations

-

### CODE REVIEW

**Comment by jnaviask** [comment]
@ianrowan does CE need to get deployed prior to Commonwealth? Or does order not matter?

**Comment by ianrowan** [comment]
> @ianrowan does CE need to get deployed prior to Commonwealth? Or does order not matter?
@jnaviask 
It shouldnt matter. No CE API routes were effected or models changed. Just the specific EVM listensers in CE

**Comment by jnaviask** [comment]
@ianrowan my worry is if we deploy the changes to Commonwealth, and CE attempts to fetch URLs from Commonwealth (does it still do that? I forget how it works exactly...).

**Comment by ianrowan** [comment]
@jnaviask Ah yes. The migration should run first which would require CW to deploy first as like you said CE reads from ChainNodes.

**Comment by jnaviask** [comment]
@ianrowan in that case, if we deploy Commonwealth first, wont CE break until we deploy it to support HTTP? Meaning we should have a CE maintenance window and make sure the two are deployed in tandem?

**Comment by ianrowan** [comment]
@jnaviask 
CE shouldnt break b/c it pulls RPCs once and creates WSS connections on startup so those will persist in the old/running CE instance even after the CW DB is updated. So as long as the existing CE remains running until the new CE code is deployed there shouldn't be an issue there. 

The only scenario where we could run into trouble would be if we migrated CW and stopped and started CE without deploying the new code.

**Review by jnaviask — APPROVED** [review]

**Comment by CowMuon** [comment]
Is this a case where we literally just missed updating a single URL on the client side? Or is there a larger pattern or assumptions the client app makes here that needs to be updated.

**Comment by ianrowan** [comment]
@CowMuon This is the only part of the app i'm aware of that uses a hardcoded value(and probably shouldnt) for chainNode, and its only used for the purpose of hardcoding the form entry for Eth RPC. 

If we hard code anything it should be the database chainNode id rather than relying on string matching to determine if we need to create a new chainNode. But realistically we should always resolve the proper chainNode via chain_id

In summary DB should be the source of chainNode/replace hard coded to support future migrations similar to this one
